### PR TITLE
LTE link control: Add API for active LTE mode

### PR DESCRIPTION
**Pull request by jtguggedal**
lib: lte_lc: Add API and event to get current LTE mode

- Introduce enum for LTE mode.

- Add API to get the currently active LTE mode. If more than one
  LTE mode is enabled in the system mode configuration, this API
  can be used to know which of LTE-M and NB-IoT is currently being used
  by the modem.

- Change CEREG parsing function to handle both notifications and reads.
  Also allow to retrieve a subset of the information available in a
  CEREG message.

---

lib: lte_lc: Fix LTE preference handling 

In some cases, the LTE preference set by Kconfig could be
overwritten by the value read back from the modem during
initialization of the link controller.
This patch introduces the same scheme for handling LTE preference
as the one used to handle system mode.

---

samples: nrf9160: cloud_client: Print when LTE mode changes 

Add printing of event information when LTE mode changes.

### CODE REVIEW

**Review by tokangas — APPROVED** [review]
Looks good, only found one typo.

File: include/modem/lte_lc.h [hunk 1]
```diff
@@ -109,6 +120,14 @@ enum lte_lc_evt_type {
 	LTE_LC_EVT_EDRX_UPDATE,
 	LTE_LC_EVT_RRC_UPDATE,
 	LTE_LC_EVT_CELL_UPDATE,
+
+	/** The currently active LTE mode is updated. If a system mode that
+	 *  enables both LTE-M and NB-IoT is configured, the modem may change
+	 *  the currently active LTE mdode based on the system mode preference
```
[the remark this answers is not in the document]
Comment (jtguggedal) [reply include/modem/lte_lc.h]: Thanks, fixed now

**Review by rlubos — APPROVED** [review]
Looks good. I guess this new API function could be used in the lwm2m_client sample as well to determine the current LTE mode? (https://github.com/nrfconnect/sdk-nrf/blob/master/samples/nrf9160/lwm2m_client/src/lwm2m/lwm2m_connmon.c#L37) I'll send a patch once this PR gets in.

**Comment by jtguggedal** [comment]
> Looks good. I guess this new API function could be used in the lwm2m_client sample as well to determine the current LTE mode? (https://github.com/nrfconnect/sdk-nrf/blob/master/samples/nrf9160/lwm2m_client/src/lwm2m/lwm2m_connmon.c#L37) I'll send a patch once this PR gets in.

@rlubos Yes, it should be a good alternative there

**Review by jhirsi — APPROVED** [review]

**Review by jhn-nordic — APPROVED** [review]

**Pull request merged**